Feature/introduce seances

MonikaLukowska opened the pull request:
In this PR I introduce seances.  It covers:

- adding model, controller and views
- adding associations to movie and hall
- adding system and model tests

User can:

- create, edit and destroy seance
- lists seances grouped by movies from current date
- navigate to seances from the 6 following days

https://user-images.githubusercontent.com/56922072/191046589-052c5be9-cbf9-4f96-b8b2-5494638b3e30.mp4

MonikaLukowska committed:
feat: add Seances table and Seance model with validations

MonikaLukowska committed:
feat:add association to hall and movie

MonikaLukowska committed:
feat:add routes for seances

MonikaLukowska committed:
feat:add seances controller

MonikaLukowska committed:
feat:add views for seances

MonikaLukowska committed:
test:add Seance system and model test

MonikaLukowska committed:
test:change root to movies_path in Movie tests

baniak66 commented on app/controllers/seances_controller.rb:
```diff
@@ -0,0 +1,56 @@
+    if seance.update(seance_params)
+      redirect_to seance, notice: t('.notice')
+    else
+      render :edit
```
I think you miss locals here. Please check also `movies#update` action.

MonikaLukowska replied:
you're right, I should pass locals here as well

baniak66 commented on app/validators/availability_validator.rb:
```diff
@@ -0,0 +1,11 @@
+class AvailabilityValidator < ActiveModel::EachValidator
```
I like your custom validator, but there is an issue. When I try to edit the price of the existing seance I get `Start time not available` error.

MonikaLukowska replied:
Didn't think about it. Record should be excluded. Thanks:)

baniak66 reviewed (commented):
Nice job 👏 Just small remarks.

MonikaLukowska committed:
fix:add missing locals in edit method

MonikaLukowska committed:
fix:exclude record from validator date ranges

Including validated record in seances collection causes datetime conflict and validation failure

MonikaLukowska committed:
fix:change datetime format

MonikaLukowska committed:
ref: Group_by movie instead of movie_title

Get access to movie properities in case of future extentions. Remove redundant movie title delegation
